upcoming_changes/2709.improvement.rst

```diff
@@ -0,0 +1,3 @@
+Improve support for extension registration on kaggle: the extension should now
```

Comment:
What about replacing all this with:

Use ``importlib_metadata`` instead of ``pkg_resources`` for extensions registration to speed up the import process and making it possible to install extensions and use them without restarting the python session.